File: Greedy/2567.Minimum_Score_by_Changing_Two_Elements/C++/2567.Minimum_Score_by_Changing_Two_Elements.cpp
class Solution {
public:
    int minimizeSum(vector<int> & nums) 
    {
        //====================================//
        // Input type:                        //
        //   - vector<int> ref. (nums vector) //
        // Return type:                       //
        //   - int (result minimun score)     //
        //====================================//

        /*Initialize*/
        ///// Length of nums vector /////
        int len_nums = nums.size();

        /*Greedy-based loop traversal with sorted-operations*/
        sort(nums.begin(), nums.end()); //Sorted-operations

        int right_most1 = (nums.back() - nums[2]); //Right-most value 1
        int right_most2 = (nums[(len_nums - 2)] - nums[1]); //Right-most value 2
        int right_most3 = (nums[(len_nums - 3)] - nums[0]); //Right-most value 3

        return min(right_most1, min(right_most2, right_most3));
    }
};

File: Greedy/2567.Minimum_Score_by_Changing_Two_Elements/Python/2567.Minimum_Score_by_Changing_Two_Elements.py
class Solution(object):
    def minimizeSum(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        #===============================#
        # Greedy-based traversal method #
        #===============================#

        ############
        #Initialize
        ##### Length of nums array #####
        len_nums = len(nums)

        ####################################################
        #Greedy-based loop traversal with sorted-operations
        nums.sort() #Sorted-operations

        right_most1, right_most2, right_most3 = (nums[(-1)] - nums[2]), (nums[(-2)] - nums[1]), (nums[(-3)] - nums[0]) #Right-most values (i.e. 1, 2, 3)

        return min(right_most1, right_most2, right_most3)
